panic: returned a non-nil result for a "initialize" Request without an ID

You can make a Go MCP server crash by sending a JSON-RPC request without an id field.

**To Reproduce**

Steps to reproduce the behavior:
1. Start an MCP Server
    * `go run github.com/modelcontextprotocol/go-sdk/examples/hello`
1. Send a JSON-RPC without an id field.
    ```{ "jsonrpc": "2.0", "method": "initialize", "params": { "protocolVersion": "2025-03-26", "capabilities": {}, "clientInfo": { "name": "example", "version": "0" } } }```
1. The same request succeeds when an "id" field is provided
   ```{ "jsonrpc": "2.0", "id": 1, "method": "initialize", "params": { "protocolVersion": "2025-03-26", "capabilities": {}, "clientInfo": { "name": "example", "version": "0" } } }```

**Expected behavior**
The request should fail, but not crash the MCP server.

**Logs**

```
$ go run github.com/modelcontextprotocol/go-sdk/examples/hello
{ "jsonrpc": "2.0", "method": "initialize", "params": { "protocolVersion": "2025-03-26", "capabilities": {}, "clientInfo": { "name": "example", "version": "0" } } }
read: {"jsonrpc":"2.0","method":"initialize","params":{"protocolVersion":"2025-03-26","capabilities":{},"clientInfo":{"name":"example","version":"0"}}}
panic: jsonrpc2: (jsonrpc2.HandlerFunc)(0x2b85d0) returned a non-nil result for a "initialize" Request without an ID

goroutine 22 [running]:
github.com/modelcontextprotocol/go-sdk/internal/jsonrpc2.(*Connection).internalErrorf(0x40000b3d40, {0x37eb8f?, 0x30?}, {0x4000027ef8?, 0x400007c030?, 0x0?})
        /home/jraviles/projects/pkg/mod/github.com/modelcontextprotocol/go-sdk@v0.2.0/internal/jsonrpc2/conn.go:763 +0xfc
github.com/modelcontextprotocol/go-sdk/internal/jsonrpc2.(*Connection).processResult(0x40000b3d40, {0x315360, 0x40000b7f08}, 0x40000db3e0, {0x3075e0, 0x40000700c0}, {0x0, 0x0})
        /home/jraviles/projects/pkg/mod/github.com/modelcontextprotocol/go-sdk@v0.2.0/internal/jsonrpc2/conn.go:707 +0x604
github.com/modelcontextprotocol/go-sdk/internal/jsonrpc2.(*Connection).handleAsync(0x40000b3d40)
        /home/jraviles/projects/pkg/mod/github.com/modelcontextprotocol/go-sdk@v0.2.0/internal/jsonrpc2/conn.go:662 +0x128
created by github.com/modelcontextprotocol/go-sdk/internal/jsonrpc2.(*Connection).acceptRequest.func2 in goroutine 20
        /home/jraviles/projects/pkg/mod/github.com/modelcontextprotocol/go-sdk@v0.2.0/internal/jsonrpc2/conn.go:624 +0x14c
exit status 2
```

--- a/mcp/transport.go
+++ b/mcp/transport.go
@@ -10,6 +10,7 @@ import (
 	"errors"
 	"fmt"
 	"io"
+	"log"
 	"net"
 	"os"
 	"sync"
@@ -120,6 +121,7 @@ func connect[H handler](ctx context.Context, t Transport, b binder[H]) (H, error
 		OnDone: func() {
 			b.disconnect(h)
 		},
+		OnInternalError: func(err error) { log.Printf("jsonrpc2 error: %v", err) },
 	})
 	assert(preempter.conn != nil, "unbound preempter")
 	h.setConn(conn)

